src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -1041,22 +1048,336 @@ public static boolean xor(final boolean x, final boolean y) {
+                        int indelSize = refIndexBack - matchIndex + 1 - matchingBases;
+                        if(indelSize <= deletion.getIndelSize()){
+                            deletion.setAlignmentOffset(alignmentOffset);
+                            deletion.setMatchingBases(matchingBases);
+                            deletion.setIndelSize(indelSize);
+                            deletion.setIndelType(false);
+                        }
+                    }
+                }
+                //************************************
+
+                int queryIndexFront = queryIndexBack - 1;
+                //set bound. this ensures that you don't look for insertions greater than the maxInsertionSize
+                int queryFrontBound = queryIndexBack - maxInsertionSize;
+
+                //repeat the process, but this time you're determining the bases on the left side of the insertion
+                while(queryIndexFront >= queryFrontBound){
+
+                    int refIndexFront = refIndexBack;
+
+                    //look for match
+                    while(reference[refIndexFront] != query[queryIndexFront] && queryIndexFront >= queryFrontBound){
+                        queryIndexFront--;
+
+                        if(queryIndexFront < queryFrontBound){
+                            break;
+                        }
+                    }
+
+                    if(queryIndexFront < queryFrontBound){
+                        break;
+                    }
+
+                    Queue<Integer> overlap2 = new LinkedList<>();
+                    overlap2.add(queryIndexFront);
+                    byte repeatLetter2 = query[queryIndexFront];
+                    boolean inListOfStartingPositions2 = false;
+                    int refIndex = refIndexFront;
+
+                    while(!overlap2.isEmpty()){
+
+                        queryIndexFront = overlap2.remove();
+                        if(queryIndexFront < queryFrontBound){
+                            break;
+                        }
+                        refIndexFront = refIndex;
+
+                        //iterate until mismatch or insertion is found
+                        while(queryIndexFront >= 0 && refIndexFront >= 0 && reference[refIndexFront] == query[queryIndexFront]){
+                            queryIndexFront--;
+                            refIndexFront--;
+
+                            //you've reached the end of the query, which means you have an insertion
+                            if(queryIndexFront == -1){
+                                int alignmentOffset = refIndexFront + 1;
+                                int matchingBases = refIndexBack - alignmentOffset + 1;
+                                int indelSize = queryIndexBack - matchingBases + 1;
+                                if(indelSize <= insertion.getIndelSize()){
+                                    insertion.setAlignmentOffset(alignmentOffset);
```

Comment:
make these Indel objects immutable and just construct new ones if you find a shorter indel.